Drivers/Inc/Minions.h

```diff
@@ -1,116 +1,48 @@
-/* Copyright (c) 2020 UT Longhorn Racing Solar */
-
-#ifndef MINIONS_H
-#define MINIONS_H
-#include "BSP_SPI.h"
+#ifndef MINION_H
+#define MINION_H
+#include <stdbool.h>
 #include "BSP_GPIO.h"
-#include "common.h"
-#include "config.h"
-#include "GPIOExpander.h"
-#include "Tasks.h"
-
-#define SPI_CS          GPIO_Pin_4
-#define LIGHTS_PORT     PORTC
-#define HEADLIGHT_PIN   GPIO_Pin_6
-#define BRAKELIGHT_PIN  GPIO_Pin_7
-#define RIGHT_BLINK_PIN GPIO_Pin_8
-#define LEFT_BLINK_PIN  GPIO_Pin_9
-
-typedef enum {
-     HZD_SW=0, 
-     CRUZ_EN, 
-     REV_SW,
-     FOR_SW,
-     HEADLIGHT_SW,
-     LEFT_SW,
-     RIGHT_SW,
-     REGEN_SW,
-     CRUZ_ST,
-     IGN_1,
-     IGN_2
-} switches_t;
-
-// Ordered by pin of minion board
-//TODO: Double check that this ordering remains valid
-typedef enum {  
-    A_CNCTR = 0,
-    M_CNCTR,
-    CTRL_FAULT,
-    LEFT_BLINK,
-    RIGHT_BLINK,
-    Headlight_ON,
-    BrakeLight,
-    RSVD_LED
-} light_t;
-
-/**
- * @brief   Initializes all Lights and Switches
- * @param   None
- * @return  None
- */ 
-void Minions_Init(void);
-
-/**
- * @brief   Reads the current state of 
- *          the specified switch
- * @param   sw the switch to read
- * @return  State of the switch (ON/OFF)
- */ 
-State Switches_Read(switches_t sw);
-
-/**
-* @brief   Read the state of the lights
-* @param   light Which Light to read
-* @return  returns state enum which indicates ON/OFF
-*/ 
-State Lights_Read(light_t light);
-
-/**
- * @brief   Sends SPI messages to read switches values. Also reads from GPIO's for 
- *          ignition switch values
- */ 
-void Switches_UpdateStates(void);
 
-/**
- * @brief   Set light to given state
- * @param   light which light to set
- * @param   state what state to set the light to
- * @return  void
- */
-void Lights_Set(light_t light, State state);
-
-/**
-* @brief   Read the lights bitmap
-* @return  returns uint16_t with lights bitmap
-*/ 
-uint16_t Lights_Bitmap_Read();
-
-/**
- * @brief Toggles a light. Should be used only after Toggle_Enable has been called for this light so that we are accurately tracking the enabled and disabled lights
- * @param light Which light to toggle
-*/
-void Lights_Toggle(light_t light);
-
-/**
-
- * @brief   Set light toggling
- * @param   light Which light to enable toggling for
- * @param   state State to set toggling
- * @return  void
+typedef enum{
+    //inputs
+    REGEN_SW,
+    FOR_SW,
+    REV_SW,
+    CRUZ_EN,
+    CRUZ_ST,
+    //output
+    BRAKELIGHT, 
+    MINIONPIN_NUM,
+} MinionPin_t; 
+
+typedef struct PinInfo{
+    uint16_t pinmask;
```

Comment:
Calling this a Pin Mask doesn't make a ton of sense, I'd call it a `pinNumber` instead